Kernel/Arch/riscv64/SBI.h

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright (c) 2023, Sönke Holz <sholz8530@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Error.h>
+#include <AK/Format.h>
+
+// Documentation about the SBI:
+// RISC-V Supervisor Binary Interface Specification (https://github.com/riscv-non-isa/riscv-sbi-doc)
+
+namespace Kernel::SBI {
+
+// Chapter 3. Binary Encoding
+enum class SBIError : long {
+    // SBI_SUCCESS: Completed successfully
+    Success = 0,
+    // SBI_ERR_FAILED: Failed
+    Failed = -1,
+    // SBI_ERR_NOT_SUPPORTED: Not supported
+    NotSupported = -2,
+    // SBI_ERR_INVALID_PARAM: Invalid parameter(s)
+    InvalidParam = -3,
+    // SBI_ERR_DENIED: Denied or not allowed
+    Denied = -4,
+    // SBI_ERR_INVALID_ADDRESS: Invalid address(s)
+    InvalidAddress = -5,
+    // SBI_ERR_ALREADY_AVAILABLE: Already available
+    AlreadyAvailable = -6,
+    // SBI_ERR_ALREADY_STARTED: Already started
+    AlreadyStarted = -7,
+    // SBI_ERR_ALREADY_STOPPED: Already stopped
+    AlreadyStopped = -8,
+    // SBI_ERR_NO_SHMEM: Shared memory not available
+    NoSHMEM = -9,
+};
+
+template<typename T>
+using SBIErrorOr = ErrorOr<T, SBIError>;
+
+enum class EID : i32 {
+    // Base Extension
+    Base = 0x10,
+    // Debug Console Extension ("DBCN")
+    DebugConsole = 0x4442434E,
+    // Timer Extension ("TIME")
+    Timer = 0x54494D45,
+};
+
+// Chapter 4. Base Extension (EID #0x10)
+// Required extension since SBI v0.2
+namespace Base {
+
+enum class FID : i32 {
+    GetSpecVersion = 0,
+    GetImplID = 1,
+    GetImplVersion = 2,
+    ProbeExtension = 3,
+    GetMVENDORID = 4,
+    GetMARCHID = 5,
+    GetMIMPID = 6,
+};
+
+struct SpecificationVersion {
```

Comment:
I think that this should be declared `packed`

Reply:
In general if there's a requirement that a struct have a specific size, we should probably `static_assert(sizeof(Blah) == specific_size)` to make the compiler check our assumptions. There's plenty of code around that doesn't follow best practices, which we should update :)

Reply:
The assertion is actually more important, so it should be fine IMO